pkg/appmanager/profiles.go

```diff
@@ -474,6 +474,22 @@ func (appMgr *Manager) deleteUnusedProfiles(
 							}
 						}
 					}
+					if serverProfile, ok :=
+						ing.ObjectMeta.Annotations[f5ServerSslProfileAnnotation]; ok == true {
+						var profName string
+						// Copied from above, but why are we not using 'formatIngressSslProfileName()' as the code in resourceConfig?
+						// Trim leading "/" from secret name (if it exists)
+						secretName := strings.TrimSpace(strings.TrimPrefix(serverProfile, "/"))
+						// If another slash remains, then we know a partition is in the name
+						if strings.ContainsAny(secretName, "/") {
+							profName = fmt.Sprintf("%s/%s", prof.Partition, prof.Name)
+						} else {
+							profName = prof.Name
+						}
+						if profName == secretName {
+							referenced = true
+						}
+					}
```

Comment:
A commit just went in that took this common code block and moved it into a function (also pulled in the "secrets" check from above). If you want to rebase on the latest commit, you could likely remove this chunk and use that function, as it fits with your annotation check.